src/app/components/tableView/applyFiltersAndVisibility.js

```diff
@@ -142,15 +143,28 @@ const tableOrFiltersChanged = (props, nextProps) => {
   );
 };
 
+const listToSet = (...fs) => list => {
+  const into = (set, item) => set.add(item);
+  const xform = f.compose(...fs);
+
+  return f.reduce(xform(into), new Set(), list);
+};
+
 const getSortedVisibleColumns = (columnOrdering, visibleColumns, columns) => {
   const statusColumnIndex = f.findIndex({ kind: ColumnKinds.status }, columns);
-
-  const orderVisible = f.reduce((acc, val) => {
-    if (f.contains(val.id, visibleColumns)) {
-      return f.concat(acc, [val.idx]);
-    }
-    return acc;
-  }, []);
+  const visibleColumnIDs = new Set(visibleColumns);
+  const hiddenColumnIDs = listToSet(
+    t.filter(f.prop("hidden")),
+    t.map(f.prop("id"))
+  )(columns);
+
+  const isVisibleColumnId = id =>
+    visibleColumnIDs.has(id) && !hiddenColumnIDs.has(id);
+
+  const orderVisible = t.transduceList(
```

Comment:
Was auch immer `transduceList` macht...
> Als Transduktion wird in der Genetik der Gentransfer durch Viren bezeichnet.

Wenn das ein Insider-Spezialbegriff ist, sollte man das vielleicht umbenennen, damit man aus der Blackbox-Sicht auch nachvollziehen kann, was hier passiert.

Reply:
Hier muss ich Hermann verteidigen. Ich kannte den Begriff erst auch nicht, aber Google hilft :)
https://medium.com/javascript-scene/transducers-efficient-data-processing-pipelines-in-javascript-7985330fe73d

Reply:
Thema für ein Bearly?

Reply:
Transducers sind schon fortgeschrittenes FP. Siehe auch den Headerkommentar in https://github.com/campudus/tableaux-frontend/pull/311/files#diff-185a97431495541a0e6d73aa733de0b9d35816731a9e8dbe91bcbd0fd16b5dbc